### PR TITLE
Stop restarting after 10 failed restarts [ESD-654]

### PR DESCRIPTION
**Pull request by silverjam**
+ If a service fails to restart 10 times (consecutively), don't continue
to try to restart it

+ Make shellcheck happy

### CODE REVIEW

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.1.0-develop-2018111323-4-g0397c86c
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=767806)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.1.0-develop-2018111323-4-g0397c86c&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.1.0-develop-2018111323-4-g0397c86c&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.1.0-develop-2018111323-4-g0397c86c
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.1.0-develop-2018111323-4-g0397c86c)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.1.0-develop-2018111323-4-g0397c86c/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.1.0-develop-2018111323-4-g0397c86c

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.1.0-develop-2018111323-5-gdc323691
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=768819)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.1.0-develop-2018111323-5-gdc323691&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.1.0-develop-2018111323-5-gdc323691&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.1.0-develop-2018111323-5-gdc323691
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.1.0-develop-2018111323-5-gdc323691)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.1.0-develop-2018111323-5-gdc323691/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.1.0-develop-2018111323-5-gdc323691

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.1.0-develop-2018111323-6-geef2a794
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=770338)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.1.0-develop-2018111323-6-geef2a794&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.1.0-develop-2018111323-6-geef2a794&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.1.0-develop-2018111323-6-geef2a794
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.1.0-develop-2018111323-6-geef2a794)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.1.0-develop-2018111323-6-geef2a794/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.1.0-develop-2018111323-6-geef2a794

**Comment by silverjam** [comment]
*TODO:* Need to change SmoothPose to not attempt to connect it's nav_daemon ports (or maybe just create them but don't do anything with them on the Piksi side) before we can merge this.

File: package/common_init/overlay/usr/bin/service_stopped [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+# shellcheck disable=SC1091,SC2169
+
+set -e
+
+name=$1             ; shift
+failed_restarts=$1  ; shift
+run_status=$1       ; shift
+
+# How long to sleep in between service restarts
+restart_sleep_time=0.5
```
[the remark this answers is not in the document]
Comment (silverjam) [reply package/common_init/overlay/usr/bin/service_stopped]: This is 500ms!  An exponential back-off sounds interesting, but I think this is complicated by services like SmoothPose-- if this crashes, you'd want this to restart pretty quickly, 500ms is already a long time to not have INS.

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.1.0-develop-2018120501-9-g8a7816d1

These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.

### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=799285)

### gnss-analysis results
At least one run must complete for these links to have data.

#### passfail"
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2018120501-9-g8a7816d1&build_type=buildroot_pr&display_type=table)

#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend-post&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2018120501-9-g8a7816d1&build_type=buildroot_pr&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.1.0-develop-2018120501-9-g8a7816d1
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.1.0-develop-2018120501-9-g8a7816d1)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.1.0-develop-2018120501-9-g8a7816d1/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.1.0-develop-2018120501-9-g8a7816d1

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.1.0-develop-2018120522-9-gc3412f73

These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.

### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=800576)

### gnss-analysis results
At least one run must complete for these links to have data.

#### passfail"
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2018120522-9-gc3412f73&build_type=buildroot_pr&display_type=table)

#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend-post&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2018120522-9-gc3412f73&build_type=buildroot_pr&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.1.0-develop-2018120522-9-gc3412f73
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.1.0-develop-2018120522-9-gc3412f73)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.1.0-develop-2018120522-9-gc3412f73/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.1.0-develop-2018120522-9-gc3412f73

File: package/common_init/overlay/usr/bin/service_stopped (hunk 1 above)
[the remark this answers is not in the document]
Comment (pmiettinen) [reply package/common_init/overlay/usr/bin/service_stopped]: True, I think the 5 second comment came from 10*0.5s.

**Pull request merged**